xclim/indices.py

```diff
@@ -669,14 +669,18 @@ def growing_season_length(tas, thresh=5.0, window=6, freq='YS'):
     i = xr.DataArray(np.arange(tas.time.size), dims='time')
     ind = xr.broadcast(i, tas)[0]
 
-    c = ((tas > thresh + K2C) * 1).rolling(time=window).sum(dim='time')
+    c = ((tas > thresh + K2C) * 1).rolling(time=window).sum()
     i1 = ind.where(c == window).resample(time=freq).min(dim='time')
 
     # Resample sets the time to T00:00.
     i11 = i1.reindex_like(c, method='ffill')
 
     # TODO: Adjust for southern hemisphere
-    i2 = ind.where(c == 0).where(tas.time.dt.month >= 7)
+
+    #i2 = ind.where(c == 0).where(tas.time.dt.month >= 7)
+    # add check to make sure indice of end of growing season is after growing season start
+    i2 = ind.where((c==0) & (ind > i11)).where(tas.time.dt.month >= 7)
```

Comment:
Can we get rid of the month>7 constraint? It won't work in the southern hemisphere. 
Here is a reference to Copernicus's algorithm http://urbansis.climate.copernicus.eu/growing-season-length/

Reply:
la référence à Juillet est clairement mauvaise pour les points au sud. Par contre je vois mal comment on peut se sauver d'un traitement différent pour les deux hémisphères (via un ajustement à la valeur de la latitude). 

Aussi, pour l'hémisphère sud il faudrait voir quelle année on assigne à la valeur de GSL. Celle du début ou de la fin? Sans compter que l'année choisie pourrait ne pas correspondre à la GS identifiée (p. ex. si la GS est contenue avant ou après le 1er janvier).

Reply:
@huard in the copernicus case (i.e. no constraint of july 1) is seems possible that we would get false ends to the season in some cases.  E.g. the start date is followed by a late spring cold snap and we end up getting an end to our season before it even really begins?

Reply:
What if we look for i1 as we currently do, and look for i2 the same way but going backward in time from the end of the year ? If it works, we could raise the issue with the guys of ICCLIM and propose this fix.

Reply:
I think the backwards version would get the 'last' period of 5 consecutive days > thresh and not the first?

Reply:
Southern hemisphere causes major headache as we would need to resample the data before doing anything right?

Reply:
I'm confused. 
What about
```
i1 = ind.where(c == window).resample(time=freq)
i_hot_min = i1.min(dim='time')
i_hot_max = i1.max(dim='time')
``` 
and look for the first `i2` larger than `i_hot_max` ?

Reply:
Would we still not have to resample lat<0 as AS-AUG and lat >0 as YS in order to capture the growing season correctly?

Reply:
Also following the definition the end of season is not i_hot_max (last of the year) but the first occurence below after start date. Taking the 'max' would includes any small extra occurrences (redoux) at end of season.

Reply:
> Would we still not have to resample lat<0 as AS-AUG and lat >0 as YS in order to capture the growing season correctly?

This is assuming someone would want to run a calculation on global data of course.  Do we need to support this possibility?

Reply:
My guess is that yes, that would be the mid-term objective.

Reply:
Let's keep it simple then. We simply impose that the end of the season is after the start. If you get weird results, increase the window size.

Reply:
Ok. Short term I think this is the simpler way to go.